### ISSUE TITLE
Static linking licensing issues should be mentioned

### ISSUE THREAD

**Issue by glandium** [indygreg/PyOxidizer#1]
Even if in passing, the README should at least mention that the license of everything that is statically linked needs to be observed very carefully because some licenses require very specific things when you distribute software, which you end up doing when statically linking (which you're not doing that when dynamically linking)

**Comment by gpshead** [comment]
Yep. A classic example of this issue is readline which, while a library, is GPL. Not LGPL. So it infects the whole binary if linked in - a problem for non-GPLed applications. But most things do not _need_ readline. `libedit` is a viable friendly license alternative that can be used instead of `libreadline` if so.

**Comment by glandium** [comment]
Another example is LGPL. LGPL actually has annoying requirements for redistribution, which is why all LGPL code in Firefox is statically linked ... in a _separate_ dynamic library.

**Comment by indygreg** [comment]
I mention this at https://github.com/indygreg/python-build-standalone#licensing but forgot to carry over the prominent notice to this repo.

The licensing issue is on my mind: I had already thought of a required config option to denote license compatibility. That would automatically strip out GPL code if necessary. One of the reasons `python-build-standalone` distributes object files in addition to the final binaries is so we can link a new binary without certain parts. I haven't yet implemented this functionality though.

Thanks for the suggestion to use `libedit` over `readline` - I'll have to look into that!
